test/fixtures/db.py

```diff
@@ -317,6 +317,94 @@ def create_link_scenario(
     return dag
 
 
+@register_scenario("alt_dedupe")
+def create_alt_dedupe_scenario(
+    backend: MatchboxDBAdapter,
+    warehouse_engine: Engine,
+    n_entities: int = 10,
+    seed: int = 42,
+) -> TestkitDAG:
+    """Create a TestkitDAG scenario with two alternative dedupers."""
+    dag = TestkitDAG()
+
+    # Create linked sources
+    company_name_feature = FeatureConfig(
+        name="company_name", base_generator="company"
+    ).add_variations(SuffixRule(suffix=" UK"))
+
+    foo_a_tkit_source = SourceTestkitParameters(
+        name="foo_a",
+        engine=warehouse_engine,
+        features=(company_name_feature,),
+        drop_base=False,
+        n_true_entities=n_entities,
+        repetition=1,
+    )
+
+    linked = linked_sources_factory(source_parameters=(foo_a_tkit_source,))
+    dag.add_source(linked)
+
+    # Write sources to warehouse
+    _testkitdag_to_location(warehouse_engine, dag)
+
+    # Index sources in backend
+    for source_testkit in dag.sources.values():
+        backend.index(
+            source_config=source_testkit.source_config,
+            data_hashes=source_testkit.data_hashes,
+        )
+
+    # Create and add deduplication models
+    for testkit in dag.sources.values():
+        source = testkit.source_config
+        model_name1 = f"dedupe.{source.name}"
+        model_name2 = f"dedupe2.{source.name}"
+
+        # Query the raw data
+        source_query = backend.query(source=source.name)
+
+        # Build model testkit using query data
+        model_testkit1 = query_to_model_factory(
+            left_resolution=source.name,
+            left_query=source_query,
+            left_keys={source.name: "key"},
+            true_entities=tuple(linked.true_entities),
+            name=model_name1,
+            description=f"Deduplication of {source.name}",
+            prob_range=(0.5, 1.0),
+            seed=seed,
+        )
+
+        model_testkit2 = query_to_model_factory(
+            left_resolution=source.name,
+            left_query=source_query,
+            left_keys={source.name: "key"},
+            true_entities=tuple(linked.true_entities),
+            name=model_name2,
+            description=f"Deduplication of {source.name}",
+            prob_range=(0.5, 1.0),
+            seed=seed,
+        )
+
+        assert model_testkit1.probabilities.num_rows > 0
+        assert model_testkit1.probabilities == model_testkit2.probabilities
+
+        for model, threshold in zip(
+            [model_testkit1, model_testkit2], [50, 75], strict=True
+        ):
```

Comment:
Optional, but I personally find `zip` very unnerving and avoid it if I can. This relies less on us happening to line things up and is much clearer on what's being iterated over:

```suggestion
        for model, threshold in (
            (model_testkit1, 50),
            (model_testkit2, 75),
        ):
```